plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -0,0 +1,220 @@
+            }
+            else {
+                splitColumns = dataFilesColumns;
+            }
+            deltaLakePageSource = createDeltaLakePageSource(split);
+            currentVersionAsBlock = nativeValueToBlock(INTEGER, split.currentVersion());
+            currentVersionCommitTimestamp = split.currentVersionCommitTimestamp();
+        }
+        if (fileType == CDF_FILE) {
+            return processCdfFile();
+        }
+        return processDataFile();
+    }
+
+    private TableFunctionProcessorState processCdfFile()
+    {
+        Page page = deltaLakePageSource.getNextPage();
+        if (page != null) {
+            int filePageColumns = page.getChannelCount();
+            Block[] resultBlock = new Block[filePageColumns + NUMBER_OF_ADDITIONAL_COLUMNS_FOR_CDF_FILE];
+            for (int i = 0; i < filePageColumns; i++) {
+                resultBlock[i] = page.getBlock(i);
+            }
+            resultBlock[filePageColumns] = RunLengthEncodedBlock.create(
+                    currentVersionAsBlock, page.getPositionCount());
+            resultBlock[filePageColumns + 1] = RunLengthEncodedBlock.create(
+                    nativeValueToBlock(TIMESTAMP_TZ_MILLIS, packDateTimeWithZone(currentVersionCommitTimestamp, UTC_KEY)), page.getPositionCount());
+            return TableFunctionProcessorState.Processed.usedInputAndProduced(new Page(page.getPositionCount(), resultBlock));
+        }
+        return FINISHED;
+    }
+
+    private TableFunctionProcessorState processDataFile()
+    {
+        Page page = deltaLakePageSource.getNextPage();
+        if (page != null) {
+            int filePageColumns = page.getChannelCount();
+            Block[] blocks = new Block[filePageColumns + NUMBER_OF_ADDITIONAL_COLUMNS_FOR_DATA_FILE];
+            for (int i = 0; i < filePageColumns; i++) {
+                blocks[i] = page.getBlock(i);
+            }
+            blocks[filePageColumns] = RunLengthEncodedBlock.create(
+                    nativeValueToBlock(VARCHAR, utf8Slice("insert")), page.getPositionCount());
+            blocks[filePageColumns + 1] = RunLengthEncodedBlock.create(
+                    currentVersionAsBlock, page.getPositionCount());
+            blocks[filePageColumns + 2] = RunLengthEncodedBlock.create(
+                    nativeValueToBlock(TIMESTAMP_TZ_MILLIS, packDateTimeWithZone(currentVersionCommitTimestamp, UTC_KEY)), page.getPositionCount());
+            return TableFunctionProcessorState.Processed.usedInputAndProduced(new Page(page.getPositionCount(), blocks));
+        }
+        return FINISHED;
+    }
+
+    private DeltaLakePageSource createDeltaLakePageSource(TableChangesSplit split)
+    {
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        TrinoInputFile inputFile = fileSystem.newInputFile(split.path());
+        Map<String, Optional<String>> partitionKeys = split.partitionKeys();
+        List<String> partitionValues = new ArrayList<>();
```

Comment:
`partitionValues` is unused?

Reply:
oh, right i recently removed usage of it, thanks